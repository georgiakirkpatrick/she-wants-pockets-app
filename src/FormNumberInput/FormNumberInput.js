import React from 'react'
import './FormNumberInput.css'

const FormNumberInput = props => {
    return (
        <div className='FormNumber'>
            <label htmlFor={props.id}>{props.prompt}</label>
            <input
                type='number' 
                id={props.id} 
                name={props.name} 
                value={props.currentValue} 
                onChange={props.handleChange} 
            /> %
        </div>
    )
}

FormNumberInput.defaultProps = {
    id: 1,
    prompt: '',
    name: '',
    currentValue: 1,
    handleChange: () => {}
}

export default FormNumberInput