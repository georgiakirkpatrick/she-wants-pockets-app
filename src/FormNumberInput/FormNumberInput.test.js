import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'
import FormNumberInput from './FormNumberInput'

describe('FormNumberInput', () => {
    it ('renders without crashing', () => {
        const div = document.createElement('div')
        ReactDOM.render(<FormNumberInput />, div)
        ReactDOM.unmountComponentAtNode(div)
    })

    it ('render the UI as expected', () => {
        const tree = renderer
            .create(<FormNumberInput />)
            .toJSON()
        expect(tree).toMatchSnapshot()
    })
})