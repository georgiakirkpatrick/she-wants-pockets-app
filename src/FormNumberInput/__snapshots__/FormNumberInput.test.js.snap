// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormNumberInput render the UI as expected 1`] = `
<div
  className="FormNumber"
>
  <label
    htmlFor={1}
  >
    
  </label>
  <input
    id={1}
    name=""
    onChange={[Function]}
    type="number"
    value={1}
  />
   %
</div>
`;
